### PR TITLE
fix: broken release candidate workflow

### PR DESCRIPTION
**Pull request by sthaha**
This PR fixes the broken candidate release workflow by incorporating
the candidate release workflow into the main `release` workflow. This
modifies the graphs as follows
```
e2e-test -> <release-pr?> ┳━ yes ━▸  create-gh-pre-release -> publish-olm-candidate
                          ┃
                          ┗━ no ━▸  publish-olm-development
```

It additionally (in a seperate commit) also modifies the .versionrc to also
include changes to deploy/package-operator when running `make initiate-release`

Signed-by: sthaha@redhat.com

### CODE REVIEW

**Comment by sthaha** [comment]
https://github.com/sthaha/observability-operator/actions/runs/5409715783 is where I am testing the PR by creating a  candidate release in my fork.

File: .github/olm-publish/action.yaml [hunk 1]
```diff
@@ -52,7 +50,7 @@ runs:
       # make initiate-release conflicts with the previous bundle created for
       # development branch, so the version in main takes precedence over the
       # one in olm-catalog branch
-      git merge -Xours origin/olm-catalog  -m "ci: merge olm-catalog branch"
+      git merge --allow-unrelated-histories -Xours origin/olm-catalog  -m "ci: merge olm-catalog branch"
```
Comment (sthaha) [inline .github/olm-publish/action.yaml:53]: note: this was required to get my CI passing - 
See failure - https://github.com/sthaha/observability-operator/actions/runs/5408840544/jobs/9828324412

**Comment by sthaha** [comment]
This need further testing because the `git merge` failed in my CI :(

**Comment by sthaha** [comment]
Tested here: https://github.com/sthaha/observability-operator/actions/runs/5452043074 
And did publish `-rc` images - https://quay.io/repository/sthaha/observability-operator-bundle?tab=tags

**Review by simonpasquier — APPROVED** [review]

**Review by jan--f — APPROVED** [review]
Thx!

**Pull request merged**